src/commoncode/functional.py

```diff
@@ -25,7 +25,15 @@
 from __future__ import absolute_import, print_function
 
 import functools
-from types import ListType, TupleType, GeneratorType
+
+# Python 2 and 3 support
+try:
+    # Python 2
+    from types import ListType, TupleType, GeneratorType
+except ImportError, NameError:
+    from types import GeneratorType
+    ListType = list
+    TupleType = tuple
```

Comment:
that's clean... but could we instead just use list.tuple everywhere and only import GeneratorType on both py2 and 3?